apps/google-analytics-4/lambda/src/middlewares/setSentryContext.ts

```diff
@@ -0,0 +1,33 @@
+import * as Sentry from '@sentry/node';
+import { NextFunction, Request, Response } from 'express';
+
+export const ContentfulContextHeaders = [
+  'X-Contentful-App',
+  'X-Contentful-ContentType',
+  'X-Contentful-Entry',
+  'X-Contentful-Environment',
+  'X-Contentful-EnvironmentAlias',
+  'X-Contentful-Field',
+  'X-Contentful-Location',
+  'X-Contentful-Organization',
+  'X-Contentful-Space',
+  'X-Contentful-User',
+];
+
+export const setSentryContext = (req: Request, _res: Response, next: NextFunction) => {
+  Sentry.setUser({ id: req.header('X-Contentful-User') as Sentry.User['id'] });
+
+  const contentfulContextHeaders = Object.keys(req.headers).filter((key) => {
+    const downcasedHeaders = ContentfulContextHeaders.map((h) => h.toLowerCase());
```

Comment:
Small thing but we're rebuilding this list of downcasedHeaders on every iteration of the loop. Instead let's build it once outside the loop.